collector/lib/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+set(COLLECTOR_LIB_SRC_FILES
```

Comment:
Are you sure about switching from `*.cpp` to the exact list of files? Might be coming as a surprise for unprepared developers.

Reply:
I just prefer the explicit way of listing source files with cmake, (I believe this is also the recommended way to do things, since globbing `*.cpp` can cause things like new files not being added to already configured builds), but you are probably right that this is not the place to make this change, I'll roll it back and we can circle back to it at a later point.